### DISCUSSION TITLE
Option to disable Sieve

### DISCUSSION THREAD

**Discussion by timmetcalfe** [pdevito3/craftsman#65] [Ideas]
Paul,

Great project. I was intrigued by all the filtering options in Swagger and then discovered Sieve thanks to you. It looks great but I use the OData for Asp.Net Core feature so the option to disable Sieve would be worthwhile in that scenario.

I want to make a donation to support your work but couldn't see a custom, one-off option. It's only for about $35 so don't go to too much trouble over it.

Thank you

Tim

**Comment by pdevito3** [comment]
Hey! 

Yeah, filters are an interesting topic. I added sieve out of the gate given how much it brings to the table, but I've been bouncing around the idea of gutting it for an alternative at some point. I need to come up with the right strategy here though.

To your point, I'll certainly add it to my list and look into it a bit, but that should be a reasonable enough add. Good suggestion!

As far as the donation, I greatly appreciate it! I updated [my sponsor tab](https://github.com/sponsors/pdevito3) to allow for one offs, but it certainly isn't necessary :-)

**Comment by timmetcalfe** [comment]
Donation made Paul.

I can recommend the OData approach. It all just works! Some info:
https://devblogs.microsoft.com/odata/supercharging-asp-net-core-api-with-odata/

**Reply by pdevito3** [reply]
That's amazing, thank you so much! 

I've seen the OData lib before and it's definitely on my shortlist! Biggest concern I saw last I looked at it was it seemed more global without the ability to control settings per entity (or even endpoint). I need to take a closer look regardless, so we'll see.

Regardless, thanks again! Really do appreciate it.
